src/ape/api/accounts.py

```diff
@@ -95,8 +95,15 @@ def call(self, txn: TransactionAPI, send_everything: bool = False) -> ReceiptAPI
         if send_everything:
             if txn.max_fee is None:
                 raise TransactionError(message="Max fee must not be None.")
-
-            txn.value = self.balance - txn.max_fee
+            if not txn.gas_limit or txn.gas_limit is None:
+                raise TransactionError(message="The txn.gas_limit is not set.")
+            else:
+                txn.value = self.balance - (txn.max_fee * txn.gas_limit)
+            if txn.value <= 0:
+                raise ValueError(
+                    f"Sender does not have enough to cover transaction value and gas: \
+                    {txn.max_fee * txn.gas_limit}"
+                )
```

Comment:
Just a little nit on style
```suggestion
            elif txn.value <= 0:
                raise ValueError(
                    f"Sender does not have enough to cover transaction value and gas: \
                    {txn.max_fee * txn.gas_limit}"
                )
            else:
                txn.value = self.balance - (txn.max_fee * txn.gas_limit)
```

Reply:
The nitpick: it's sometimes difficult to tell if you have one large if condition followed by the second that the two blocks are unrelated. It's often between to combine them into one larger `if..elif...else` statement this way if you can (or add extra spacing to delineate if you can't)

Reply:
I think the problem here is that the `txn.value` needs to be calculated before the `elif txn.value<= 0`

Reply:
I updated this section of code so that it is passing checks. Can you take another look and see if you're satisfied?